### PR TITLE
Add support for per token precision inside currency input

### PR DESCRIPTION
**Pull request by lazarmitic**

### CODE REVIEW

File: packages/tempus-client_v3/src/components/DepositModal/DepositModal.tsx [hunk 1]
```diff
@@ -12,19 +12,22 @@ import { ActionButtonState, ChartDot, SelectableChartDataPoint, PercentageDateCh
...
 }
 
 const DepositModal: FC<DepositModalProps> = props => {
-  const { open, onClose, inputPrecision, usdRates, poolStartDate, maturityTerms, chainConfig } = props;
+  const { tokens, open, onClose, poolStartDate, maturityTerms, chainConfig } = props;
   const [balance, setBalance] = useState(new Decimal(100)); // TODO: load balance for selected token
   const [maturityTerm, setMaturityTerm] = useState(maturityTerms[0]);
   const [amount, setAmount] = useState(new Decimal(0));
-  const [currency, setCurrency] = useState(Array.from(usdRates.keys())[0]);
+  const [currency, setCurrency] = useState(tokens[0].ticker);
```
Comment (danieledamiani) [inline packages/tempus-client_v3/src/components/DepositModal/DepositModal.tsx:30]: if `tokens` array is empty this will fail

**Review by iamvukasin — DISMISSED** [review]

**Review by danieledamiani — DISMISSED** [review]
a couple of comments

File: packages/tempus-client_v3/src/components/DepositModal/DepositModal.tsx (hunk 1 above)
Comment (lazarmitic) [reply packages/tempus-client_v3/src/components/DepositModal/DepositModal.tsx:27]: Resolved by requiring at least one element to be provided.

File: packages/tempus-client_v3/src/components/shared/CurrencyInput/CurrencyInput.tsx [hunk 2]
```diff
@@ -10,29 +10,32 @@ import './CurrencyInput.scss';
 const inputPercentages = [25, 50, 75, 100];
 
 export interface CurrencyInputProps {
-  precision: number;
   maxAmount: Decimal;
-  usdRates: Map<Ticker, Decimal>;
+  tokens: {
+    precision: number;
+    ticker: Ticker;
+    rate: Decimal;
```
[the remark this answers is not in the document]
Comment (lazarmitic) [reply packages/tempus-client_v3/src/components/shared/CurrencyInput/CurrencyInput.tsx]: Type for this prop is now extracted to interface directory.

**Pull request merged**